### PR TITLE
Moved Build System to use Docker

### PR DESCRIPTION
**Pull request by capveg**
Removed a lot of the make workspace top level stuff.
Internal build still (currently) uses onl-mkws -- should probably update this as well to docker later.

Verified this all builds cleanly for `make onl-{x86,powerpc,kvm}` on a clean Ubuntu 14.04 system.

Ready to merge.

Reviewer: @rlane @jnealtowns

### CODE REVIEW

File: docs/Building.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    # wget -qO- https://get.docker.com/ | sh
+
+
+If you are installing on Ubuntu 14.04 or earlier:
+
+- You may have to update your kernel to 3.10+
+- Beware that `apt-get install docker` installs a dock application not docker :-)  You want the docker.io package instead.
+- Check out http://docs.docker.com/installation/ubuntulinux/ for details
+
+Consider enabling builds for non-priviledged users with:
```
Comment (rlane) [inline docs/Building.md]: Typo.

File: tools/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,121 @@
+FROM debian
```
Comment (rlane) [inline tools/docker/Dockerfile]: You might want to use a more specific tag, like `debian:7.8` or `debian:8.0`.

Comment (jnealtowns) [inline tools/docker/Dockerfile]: debian:7.8

File: docs/Building.md (hunk 1 above)
Comment (capveg) [inline docs/Building.md]: Fixed.  Appreciate the diplomacy of "typo" versus "Rob is a fool who can't spell"

File: tools/docker/Dockerfile (hunk 2 above)
Comment (capveg) [inline tools/docker/Dockerfile]: Set to debian:7.8 ; The debian 'latest' actually points to 7.4, but I just did a clean build and everything works.  Bumped the onl-build version to 1.5 to mark this change.  Thanks.

**Comment by capveg** [comment]
OK -- I think this is ready to merge.  Please let me know and thanks.

File: docs/Building.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+After all components have been built, your can build an ONL
+Software Image from those components.
+
+Adding/Removing packages from a SWI:
+------------------------------------------------------------
+
+The list of packages for a given SWI are in
+    $ONL/builds/swi/$ARCH/all/rootfs/rootfs.$ARCH	# for $ARCH specific packages
+    $ONL/builds/swi/$ARCH/all/rootfs/rootfs		# for $ARCH-independent packages
```
Comment (rlane) [inline docs/Building.md]: The Markdown output collapses these three lines into one. I think you need a blank line after the first line.

Comment (capveg) [inline docs/Building.md]: Fixed.  Thanks.

**Comment by rlane** [comment]
I reviewed just the Dockerfiles, Building.md, and the Makefile that calls docker. Those pieces LGTM.

I also ran an onl-x86 build with docker to try it out and it worked fine for me.

**Comment by capveg** [comment]
Thanks Rich for the extra testing!

@Jeff: given that people are actively having problems with the old build
system on the mailing list, I'm going to merge this late tonight unless I
hear from you that you would like more time to review it (which is fine -
this stuff is important).

Thanks,

- Rob
.

**Comment by jnealtowns** [comment]
Go ahead. We'll shake it out as necessary.

J

**Pull request merged**